sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationExec.scala

```diff
@@ -78,17 +78,25 @@ case class HiveScriptTransformationExec(
       stderrBuffer,
       "Thread-ScriptTransformation-STDERR-Consumer").start()
 
-    val outputProjection = new InterpretedProjection(input, child.output)
-
     // This nullability is a performance optimization in order to avoid an Option.foreach() call
     // inside of a loop
     @Nullable val (inputSerde, inputSoi) = ioschema.initInputSerDe(input).getOrElse((null, null))
 
+    // For HiveScriptTransformationExec, if inputSerde == null, but outputSerde != null
+    // We will use StringBuffer to pass data, in this case, we should cast data as string too.
+    val finalInput = if (inputSerde == null) {
+      input.map(Cast(_, StringType).withTimeZone(conf.sessionLocalTimeZone))
```

Comment:
This suggested `CAST` approach looks good, but we get a little behaviour changes for some types?

Reply:
> This suggested `CAST` approach looks good, but we get a little behaviour changes for some types?

To be honest, transform can't support array/map/struct etc,  now we just keep normal datatype correct is ok.
And here we handle like default LazySimpleSerde.

For input and output data serialization， add spark's own serde is the best way, and with that pr, this code can be removed.

Reply:
Ur, ... what's an answer of my question? You meant we get behaviour changes, but that' not a problem? At least, I think we need to define supported an unsupported cases, clearly (& add tests for all the cases).

Reply:
> Ur, ... what's an answer of my question? You meant we might get a behaviour change, but that's not a problem? (what's a normal type?) At least, I think we need to define supported an unsupported cases, clearly (& add tests for all the cases).

As alfozan said, here we implement a limited  data wrapper like row formatted. tread all data as string. 
https://github.com/apache/spark/pull/29085#discussion_r454298255

Reply:
> > Ur, ... what's an answer of my question? You meant we might get a behaviour change, but that's not a problem? (what's a normal type?) At least, I think we need to define supported an unsupported cases, clearly (& add tests for all the cases).
> 
> As alfozan said, here we implement a limited data wrapper like row formatted. tread all data as string.
> [#29085 (comment)](https://github.com/apache/spark/pull/29085#discussion_r454298255)

Yes; I'll follow with a PR to add native SerDe classes with complex types support (array/map/struc).